src/openassetio-core/include/openassetio/typedefs.hpp

```diff
@@ -8,8 +8,8 @@
 #include <utility>
 #include <vector>
 
+#include <openassetio/macros.hpp>
 #include <openassetio/export.h>
```

Comment:
[fix] CI linters [are complaining](https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/15619139028/job/44036846770?pr=1461).  Clang-Tidy is complaining I think because the headers are not in alphabetical order. Cpplint is complaining because it wants C headers before C++ headers.

I think just reversing the order here will work.

Reply:
Looks like you've renamed `export.h` to `export.hpp`, which doesn't exist and is failing CI.